aleph-client/src/utility.rs

```diff
@@ -6,17 +6,33 @@ use crate::{
     BlockHash, Connection,
 };
 
+/// Any object that implements retrieving various info about blocks.
 #[async_trait::async_trait]
 pub trait BlocksApi {
+    /// Returns the first block of a session.
+    /// * `session` - number of the session to query the first block from
     async fn first_block_of_session(&self, session: SessionIndex) -> Option<BlockHash>;
+
+    /// Returns hash of a given block.
+    /// * `block` - number of the block
     async fn get_block_hash(&self, block: BlockNumber) -> Option<BlockHash>;
+
+    /// Returns the most recent block from the current best chain.
     async fn get_best_block(&self) -> BlockNumber;
+
+    /// Returns the most recent block from the finalized chain.
     async fn get_finalized_block_hash(&self) -> BlockHash;
+
+    /// Returns number of a given block hash.
```

Comment:
same here `if block hash exists on the chain`

Reply:
done